KsApi/extensions/NSURLSession.swift

```diff
@@ -24,7 +24,8 @@ internal extension URLSession {
       .flatMap(.concat) { data, response -> SignalProducer<Data, GraphError> in
         guard let response = response as? HTTPURLResponse else { fatalError() }
 
-        error?.handleError(blockResponse: response, and: data)
+        /// `error` is `nil` or `handleError` returns `false`.
+        guard [nil, false].contains(error?.handleError(response: response, and: data)) else { return .empty }
```

Comment:
Ah so `.empty` will terminate right here if there is an error?

Reply:
Since `.empty` calls `completed` with no value returned to the Observer/Signal. `rac_graphDataResponse` pretends like nothing happened if there's an error or `handleError` = true while delegating error handling to `error?.handleError`.

Are we showing the captcha response to the user somewhere, or what kind of action do we perform when we call `GoToPerimeterXCaptcha`?

Reply:
Yeah so just in case the request errors out and we display some sort of error dialogue, this will just return an empty `SignalProducer` and the captcha should be displayed.